### ISSUE TITLE
Management

### ISSUE THREAD

**Issue by nickmcintyre** [JithinKS97/dynamic-learning-app#7]
I think it would be helpful to monitor and manage individual students and entire classes. Khan Academy has a nice feature that allows teachers to assign lessons and monitor students' progress (time spent, performance, etc.). One possibility is to integrate with Google Classroom and G Suite, allowing teachers to use management and content creation tools they've already mastered. I'm imaging embedding spreadsheets and slides along with simulations.

**Comment by JithinKS97** [comment]
@ashneeldas2 have some progress in this. 
1) Teachers and students can add school to their profile
2) Teachers can search a student in the same school
3) Teachers can add classes
4) Students can join the classes

I think we need more discussions as to how classroom feature can be used to enhance the learning process.

**Comment by nickmcintyre** [comment]
Classrooms could be useful for safeguarding students and streamlining lessons.

Student accounts that are only visible within a school/class community make it easy to sidestep a number of concerns. Children could always ask their parents to create a Standard account for use at home.

It may sound a little draconian, but limiting what students can do during a lesson can be a huge win for productivity. I use [GoGuardian](https://www.goguardian.com/teacher.html) regularly in our Chromebook Lab to focus students on the 2-3 websites I want them to explore. My vision for the classroom is simple: a private space where teachers curate lessons for their students so they can interact, discuss, create, fork, etc., together.

I think curation could take a couple of forms.
- Assign a single lesson for all students
- Assign a set of lessons (see #6) for students to complete at their own pace

This opens up some interesting possibilities. It'd be neat to assign my class a minimal lesson to explore, then have them explain and teach the accompanying simulation to the rest of the class. They could then critique and remix their classmates' work. If the students have learned p5.js, then they could create simulations themselves and engage in the full [Creative Learning Spiral](https://learn.media.mit.edu/lcl/resources/readings/chapter1-excerpt.pdf?pdf=ch1-en).

**Comment by JithinKS97** [comment]
The present structure of the app was made without having the idea of classroom in my mind. The idea was that teacher creates a lessonplan and programmers and teachers have discussion in the discussion forum associated with that lessonplan to create more simulations. 

But I think the idea can be extended. A simple quick tweak might be is to give an option in lessonplan creator to associate it with a class. So that,  that lessonplan and the discussion forum associated with it will be exclusively accessible to the teacher and the students. The diacussion forum and lessonplan will not be visible to the outside world.

**Comment by JithinKS97** [comment]
Let me suggest a design.

1) In the Lessonplan creator area, a button can be given called "Assign to class". When it is clicked, a modal box will be open allowing the teacher to select a classroom from the list of classes he has created.
Now the lessonplan is mapped to the classroom.

2) In the Class section in the profiles page, the students will be able to view all the lessonplans under their class. They can click on any of them to view them and they also have access to the discussion forum.

3) The students can fork the lessonplan once and all the forks will map to the same discussion forum.

These are the minimal set of features. What do you guys think?

**Comment by nickmcintyre** [comment]
I like this approach–it's straightforward and provides the essential functionality.

**Comment by JithinKS97** [comment]
The workbooks added under a class is not shown for students even though its shown for teachers. @ashneeldas2 could you please check  that?

**Comment by nickmcintyre** [comment]
What do y'all think about automatically generating and updating class-specific folders for students' lessons and workbooks? Most of the components are already in place, intuitive, and working well.

**Comment by JithinKS97** [comment]
Are you saying that just as we have organization of workbooks as folders, we can auto generate the kind of folders in the class section?

**Comment by nickmcintyre** [comment]
I actually imagined having workbooks automatically forked upon assignment and organized into a folder with the class' name, all within the existing workbook section. Something similar would apply for assigned lessons. The class section's functionality could be limited to enrollment and listing, perhaps displayed as cards with a quick description provided by the teacher.

**Comment by JithinKS97** [comment]
Can we do it like this? In the classes section, teachers can now see the workbooks under each class that he created. A button can be placed aside each of the workbook which can open up a modal box listing down all the forks created by the students in the class for that workbook. Teachers can navigate to the fork by clicking the items in the list.

**Comment by nickmcintyre** [comment]
Yep–this seems ideal for the teacher's view of the class section! Teachers can continue to organize workbooks however they choose in the workbook section.

My previous suggestion still applies for student views of these sections.
